src/blocks/recommendations/recommendations-store.tsx

```diff
@@ -6,17 +6,29 @@ import { createContext, useContext, useReducer } from '@wordpress/element';
 /**
  * Internal dependencies
  */
-import { RECOMMENDATIONS_BLOCK_ERROR, RECOMMENDATIONS_BLOCK_LOADED, RECOMMENDATIONS_BLOCK_RECOMMENDATIONS } from './constants';
+import { RecommendationsAction } from './constants';
+import { Recommendation } from './models/Recommendation';
 
-const RecommendationsContext = createContext();
+interface RecommendationState {
+	isLoaded: boolean;
+	recommendations: Recommendation[] | undefined;
```

Comment:
Is the undefined state needed? If not, maybe use an empty array in the default state. Would simplify logic.